packages/core/src/store/level-state-store.ts

```diff
@@ -41,16 +49,36 @@ export class LevelStateStore implements StateStore {
     return this.#store
   }
 
+  private _makeStore(directoryName: string): Level {
+    const storePath = path.join(this.storeRoot, directoryName)
+    if (fs) {
+      fs.mkdirSync(storePath, { recursive: true }) // create dir if it doesn't exist
+    }
+    return new LevelC(storePath)
+  }
+
   /**
-   * Open pinning service
+   * Open pinning service.
+   * Always store the pinning state in a network-specific directory.
    */
   async open(networkName: string): Promise<void> {
-    // Always store the pinning state in a network-specific directory
-    const storePath = path.join(this.storeRoot, networkName)
-    if (fs) {
-      fs.mkdirSync(storePath, { recursive: true }) // create dir if it doesn't exist
+    if (networkName == Networks.MAINNET || networkName == Networks.ELP) {
+      // If using "mainnet", check for data under an 'elp' directory first. This is because older
+      // versions of Ceramic only supported an 'elp' network as an alias for mainnet and stored
+      // state store data under 'elp' instead of 'mainnet' by mistake, and we don't want to lose
+      // that data if it exists.
+      const store = this._makeStore(Networks.ELP)
+      const pinnedStreamIds = await _listAll(store, 1)
+      if (pinnedStreamIds.length > 0) {
```

Comment:
Do we care that this will most likely stick around for a while? Should we capture this with a follow up ticket to migrate the state-store directory eventually?

Reply:
I honestly don't know when we'll be able to remove support for this TBH.  Maybe if we force the state store to always be in the DB and stop supporting the S3 state store entirely.  But otherwise we might always have nodes out there still with data in the `elp` directory.  Curious if you have any thoughts on this @oed

Reply:
That's probably fine. Don't feel strongly.